eng/mgmt/automation/generate_data.py

```diff
@@ -54,20 +55,30 @@ def sdk_automation_cadl(config: dict) -> List[dict]:
                 module = match.group(2)
 
             if not module:
-                logging.warning('[Skip] sdk-folder option {} not match format sdk/<service>/<module>'
-                                .format(sdk_folder))
+                logging.warning('[Skip] "emitter-output-dir" not match format '
+                                '{java-sdk-folder}/sdk/{service-directory-name}/<module>')
             else:
                 # cadl
                 succeeded = False
                 pwd = os.getcwd()
                 os.chdir(cadl_dir)
                 try:
-                    subprocess.run('npm install', shell=True, check=True)
+                   # install latest @azure-tools/cadl-java
+                    subprocess.run('npm install "@azure-tools/cadl-java@0.2.0"', shell=True, check=True)
+
+                    # install dependencies
+                    subprocess.run('npm install "@azure-tools/cadl-autorest@0.24.0" "@cadl-lang/eslint-config-cadl@0.5.0" "@cadl-lang/eslint-plugin@0.38.0" "@cadl-lang/library-linter@0.38.0"', shell=True, check=True)
```

Comment:
Do these dependencies not get installed by installing cadl-java? I know the cadl-autorest and cadl-csharp emitters have these as dependencies should we do the same thing for java to help eliminate these other versions we have to maintain here?

Reply:
No. None of them is used/required by cadl-java
https://github.com/Azure/autorest.java/blob/main/cadl-extension/package.json#L38-L44

And I am not sure even adding them to cadl-java would solve the problem. Maybe it would only cause dependency conflict, if package.json on spec requires "@azure-tools/cadl-autorest@0.25.0", but cadl-java requires "@azure-tools/cadl-autorest@0.24.0" (say it not yet upgraded to latest compiler; not necessary this lib, could be any one that specified in package.json on spec).

Reply:
If you require, I can try adding them to cadl-java. But no guarantee it works well though...

Reply:
We'd expect after cadl complier get stable, that cadl-java can depends on e.g. `^1.2` so that all compiler from 1.2 to 2.0 would be compatible. So that we only need to install cadl-java@latest

Reply:
I chatted with @dw511214992 the other day and we would like to try and add these as dependencies. Even if the library doesn't directly use them we should pick the set of cadl libraries that we are using to test the cadl-java package out with so that we know that everything works together correctly as a set. Having them be listed as a dependency will help us avoid having to maintain these group over versions hardcoded in various places.

Reply:
@weshaggard Sure. I will try add these here (recommended by @dw511214992) to next release of cadl-java. Then we could try it for a couple of months to see how that works out.